src/dart/ast.rs

```diff
@@ -0,0 +1,479 @@
+use enum_primitive::FromPrimitive;
+use std::iter;
+use syntax::symbol::Symbol;
+
+#[derive(Debug)]
+pub struct Library {
+    pub name: Option<LibraryName>,
+    pub items: Vec<Item>,
+}
+
+#[derive(Debug)]
+pub struct Combinator {
+    pub hide: bool,
+    pub idents: Vec<Symbol>,
+}
+
+#[derive(Debug)]
+pub struct LibraryName {
+    pub metadata: Metadata,
+    pub path: Vec<Symbol>,
+}
+
+#[derive(Debug)]
+pub struct ImportSpecification {
+    pub uri: StringLiteral,
+    pub deferred: bool,
+    pub as_ident: Option<Symbol>,
+    pub comb: Vec<Combinator>,
+}
+
+#[derive(Debug)]
+pub enum Item {
+    Import(Metadata, ImportSpecification),
+    Export(Metadata, StringLiteral, Vec<Combinator>),
```

Comment:
Maybe these two should be struct variants. We also might want to have an `Item` struct with an `ItemKind` enum holding all of these variants. E.g. `Metadata` shows up uniformly.